cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPluginFactory.java

```diff
@@ -45,11 +48,12 @@ public KafkaLogWriterPluginFactory(CConfiguration cConfig, FileMetaDataManager f
     this.namespacedLocationFactory = namespacedLocationFactory;
     this.checkpointManagerFactory = checkpointManagerFactory;
     this.impersonator = impersonator;
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
   }
 
   @Override
   public KafkaLogProcessor create() throws Exception {
     return new KafkaLogWriterPlugin(cConfig, fileMetaDataManager, checkpointManagerFactory, rootLocationFactory,
```

Comment:
The `KafkaLogWriterPlugin` has `@Inject` constructor, why we instantiate it directly? Seems like we are doing something wrong. If the `KafkaLogWriterPlugin` is always "new" directly, then remove the `@Inject` annotation.